homeassistant/components/persistent_notification/__init__.py

```diff
@@ -65,64 +85,55 @@ def async_create(
     message: str,
     title: str | None = None,
     notification_id: str | None = None,
-    *,
-    context: Context | None = None,
 ) -> None:
     """Generate a notification."""
-    if (notifications := hass.data.get(DOMAIN)) is None:
-        notifications = hass.data[DOMAIN] = {}
-
-    if notification_id is not None:
-        entity_id = ENTITY_ID_FORMAT.format(slugify(notification_id))
-    else:
-        entity_id = async_generate_entity_id(
-            ENTITY_ID_FORMAT, DEFAULT_OBJECT_ID, hass=hass
-        )
-        notification_id = entity_id.split(".")[1]
-
-    attr: dict[str, str] = {ATTR_MESSAGE: message}
-    if title is not None:
-        attr[ATTR_TITLE] = title
-        attr[ATTR_FRIENDLY_NAME] = title
-
-    hass.states.async_set(entity_id, STATE, attr, context=context)
-
-    # Store notification and fire event
-    # This will eventually replace state machine storage
-    notifications[entity_id] = {
+    notifications = _async_get_or_create_notifications(hass)
+    if notification_id is None:
+        notification_id = random_uuid_hex()
+    notifications[notification_id] = {
         ATTR_MESSAGE: message,
         ATTR_NOTIFICATION_ID: notification_id,
         ATTR_STATUS: STATUS_UNREAD,
         ATTR_TITLE: title,
         ATTR_CREATED_AT: dt_util.utcnow(),
     }
 
-    hass.bus.async_fire(EVENT_PERSISTENT_NOTIFICATIONS_UPDATED, context=context)
+    async_dispatcher_send(
+        hass,
+        SIGNAL_PERSISTENT_NOTIFICATIONS_UPDATED,
+        UpdateType.ADDED,
+        {notification_id: notifications[notification_id]},
+    )
 
 
 @callback
-@bind_hass
-def async_dismiss(
-    hass: HomeAssistant, notification_id: str, *, context: Context | None = None
-) -> None:
-    """Remove a notification."""
-    if (notifications := hass.data.get(DOMAIN)) is None:
+def _async_get_or_create_notifications(hass: HomeAssistant) -> dict[str, Notification]:
+    """Get or create notifications data."""
+    if (_notifications := hass.data.get(DOMAIN)) is not None:
+        notifications: dict[str, Any] = _notifications
+    else:
```

Comment:
You can use the `@singleton.singleton` decorator from `helpers.singleton`

Reply:
5757f167f44e880b121e3feb9f4517b7b1ceed32